php/classes/ajax/class-ajax-handler.php

```diff
@@ -232,10 +184,60 @@ public function get_external_rss_feed_progress() {
 	 * Reset external RSS feed import
 	 */
 	public function reset_external_rss_feed_progress() {
-		// @todo add nonces, add user caps check
+		self::import_security_check();
+
 		delete_option( 'ssp_external_rss' );
 		delete_option( 'ssp_rss_import' );
 		wp_send_json( 'success' );
 	}
 
+	/**
+	 * RSS feed import functions security check
+	 */
+	protected function import_security_check() {
+		try {
+			$this->user_capability_check();
+			$this->nonce_check( 'ss_podcasting_import' );
+		} catch ( \Exception $e ) {
+			$this->send_json_error( $e->getMessage() );
+		}
+	}
+
+	/**
+	 * Throws exception if nonce is not valid
+	 *
+	 * @param string $action
+	 * @param string $nonce_key
+	 *
+	 * @throws \Exception
+	 */
+	protected function nonce_check( $action, $nonce_key = 'nonce' ) {
+		if ( ! wp_verify_nonce( $_REQUEST[ $nonce_key ], $action ) ) {
+			throw new \Exception( 'Security error!' );
```

Comment:
What happens if these exceptions are thrown, does the user see an error message, or does it just fail silently?

Reply:
It alerts an error here - https://github.com/CastosHQ/Seriously-Simple-Podcasting/pull/594/files#diff-ce0a39a9329251dfc938d0642fcb5f2eb891e4d796866fa9ce107928bf20506eR104
So, the same behavior as it was before.